### ISSUE TITLE
Hidden dependencies during installation on Ubuntu

### ISSUE THREAD

**Issue by duembgen** [LCAV/pyroomacoustics#117]
I have tried installing pyroomacoustics on my Ubuntu 16.04.1 machine using pip in a clean environment, and ran into several relatively easy to fix problems. 

First I get an error that Cython needs to be installed (see Snippet 1 below). So I run `pip install Cython`. 

Then installation succeeds. 

**Update:** the following is an error coming from my virtualenv setup. 

When importing pyroomacoustics, I get the error that _bz is not found. After a little search on google I find that I should run `sudo apt-get install libbz2-dev` (see Snippet 2), which does not fix it. 

Snippet 1: 
```
Collecting pyroomacoustics
  Downloading https://files.pythonhosted.org/packages/77/b8/25af205a42b78eab360edd8199e9c953a292915941f712223b916eacc032/pyroomacoustics-0.1.23.tar.gz (269kB)
     |████████████████████████████████| 276kB 2.0MB/s 
    ERROR: Complete output from command python setup.py egg_info:
    ERROR: Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-cy5s6zj6/pyroomacoustics/setup.py", line 17, in <module>
        from Cython.Build import cythonize
    ImportError: No module named 'Cython'
    ----------------------------------------
ERROR: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-cy5s6zj6/pyroomacoustics/
```

Snippet 2:
```
ImportError                               Traceback (most recent call last)
<ipython-input-11-15a6dcad72c8> in <module>
      8 
      9 try:
---> 10     matrix_TOAs, diff_angles, W_pyramic, loc_pyramic_gt = load_experimental_data(path, verbose=True)
     11 except:
     12     print('mount the LCAV file server at ./lcav_data before runing this cell. \

~/PhD/mmgg/github/code/linear_array.py in load_experimental_data(path, plot, verbose, mic_indices, waves_indices, speaker_names, dim)
    253     if type(path) == str:
    254         __, __, __, delta, mic_array, __ = pyramic_experiment(
--> 255             path, plot, verbose)
    256     # path already contains output of pyramic_experiment.
    257     # this was added to avoid having to reload the file from the

~/PhD/mmgg/github/code/linear_array.py in pyramic_experiment(path, plot, verbose)
    134         plt.show()
    135 
--> 136     import pyroomacoustics as pra
    137     angles = list(range(0, 360, 2))
    138     spkr_list = {'middle': 0, 'low': 1, 'high': 2}

~/virtualenvs/CDMs/lib/python3.5/site-packages/pyroomacoustics/__init__.py in <module>
    135 from . import transform as realtime  # to be deprecated
    136 from . import experimental
--> 137 from . import datasets
    138 from . import bss
    139 from . import denoise

~/virtualenvs/CDMs/lib/python3.5/site-packages/pyroomacoustics/datasets/__init__.py in <module>
    140 from .base import Meta, Sample, AudioSample, Dataset
    141 from .timit import Word, Sentence, TimitCorpus
--> 142 from .cmu_arctic import CMUArcticCorpus, CMUArcticSentence, cmu_arctic_speakers
    143 from .google_speech_commands import GoogleSpeechCommands, GoogleSample

~/virtualenvs/CDMs/lib/python3.5/site-packages/pyroomacoustics/datasets/cmu_arctic.py in <module>
     40     have_sounddevice = False
     41 
---> 42 from .utils import download_uncompress
     43 from .base import Meta, AudioSample, Dataset
     44 

~/virtualenvs/CDMs/lib/python3.5/site-packages/pyroomacoustics/datasets/utils.py in <module>
     25 import os
     26 import tarfile
---> 27 import bz2
     28 
     29 try:

/usr/local/lib/python3.5/bz2.py in <module>
     20     from dummy_threading import RLock
     21 
---> 22 from _bz2 import BZ2Compressor, BZ2Decompressor
     23 
     24 

ImportError: No module named '_bz2'
```

**Comment by fakufaku** [comment]
Hi @duembgen! Thanks for the report.
I'm aware that on Linux `pip` sometimes (always ?) does not get Cython, for some reason. I do not know why and I don't think the problem is with pyroomacoustics because the dependency is in fact in [setup.py](https://github.com/LCAV/pyroomacoustics/blob/master/setup.py#L77) and there are no such problems on Mac of Windows, i.e. pip do get Cython first.
Also the `bz2` thing is mysterious. I thought this is part of the python [standard library](https://docs.python.org/3/library/bz2.html)... Did you [build you python from source](https://stackoverflow.com/questions/12806122/missing-python-bz2-module/12806325) ?

**Comment by duembgen** [comment]
Thank you for the quick reply! Makes sense for the Cython 

The bz issue might indeed be a particularity of my python installation. I did not build it from source but maybe there is a linking problem to system libraries, when I use virtual environments. I noticed that bz2 is only used in the datasets part of pyroomacoustics and that if I comment out the "import bz2" it seems to work fine as long as I do not try to load bz-compressed files.

**Comment by fakufaku** [comment]
Hum, I see that there is a `setup_requires` option for `setup.py` that might be where Cython should go since it is required to build a few things prior to install...

**Comment by duembgen** [comment]
Ok! 

Just a quick update: the bz2 error indeed disappears when I install python locally outside of the virtualenv. So indeed it is a problem with my virtualenv setup.

**Comment by duembgen** [comment]
I updated the issue accordingly

**Comment by fakufaku** [comment]
Ok, great! I'll try to fix the setup thing as this has actually happened a few times already.

**Comment by fakufaku** [comment]
Fixed in #122
